apps/shipments/models.py

```diff
@@ -341,3 +343,98 @@ def update_vault_hash(self, vault_hash, rate_limit=False):
     VALID_UNTIL = 24
     FUNDING_TYPE = FundingType.SHIP.value
     SHIPMENT_AMOUNT = 1
+
+
+class TrackingData(models.Model):
+    id = models.CharField(primary_key=True, default=random_id, max_length=36)
+    created_at = models.DateTimeField(auto_now_add=True)
+    device_id = models.CharField(blank=True, null=True, max_length=36)
+    shipment = models.ForeignKey(Shipment, on_delete=models.PROTECT, null=True)
```

Comment:
I think on_delete should probably be a CASCADE here. Also I don't think we should allow null=True for this relation? Is there a reason for that?